File: java/DataFrame/src/main/java/coyote/dataframe/S8Type.java
package coyote.dataframe;

import coyote.util.ByteUtil;

/** Type representing an signed, 8-bit value in the range of -128 to 127 */
public class S8Type implements FieldType
{
  private static final int _size = 1;

  private final static String _name = "S8";




  public boolean checkType( Object obj )
  {
    return obj instanceof Byte;//TODO: Fix Me!
  }




  public Object decode( byte[] value )
  {
    return new Byte( value[0] );//TODO: Fix Me!
  }




  public byte[] encode( Object obj )
  {
    return ByteUtil.renderUnsignedShort( (Integer)obj );//TODO: Fix Me!
  }




  public String getTypeName()
  {
    return _name;
  }




  public boolean isNumeric()
  {
    return true;
  }




  public int getSize()
  {
    return _size;
  }

}

File: csharp/DataFrame/Coyote/DataFrame/S8Type.cs
#region license
// Copyright (c) 2006 Stephan D. Cote' - All rights reserved.
// 
// This program and the accompanying materials are made available under the 
// terms of the MIT License which accompanies this distribution, and is 
// available at http://creativecommons.org/licenses/MIT/
// *
// Contributors:
//   Stephan D. Cote 
//      - Initial API and implementation
#endregion
using System;
namespace Coyote.DataFrame
{



    /// <summary>This is a numeric type equivalent to the <code>sbyte</code>type, representing an signed, 8-bit value in the range of -128 to 127.</summary>
    public class S8Type : FieldType
    {
        private const int _size = 1;

        private const string _name = "S8";



        public virtual string TypeName
        {
            get
            {
                return _name;
            }
        }




        public virtual bool IsNumeric
        {
            get
            {
                return true;
            }
        }




        public virtual int Size
        {
            get
            {
                return _size;
            }
        }




        public virtual bool CheckType(object obj)
        {
            return obj is sbyte;
        }




        public virtual object Decode(byte[] value)
        {
            return (sbyte)value[0];
        }




        public virtual byte[] Encode(object obj)
        {
            byte[] retval = new byte[1];
            retval[0] = (byte)obj;
            return retval;
        }





        public virtual string StringValue(byte[] val)
        {
            if (val == null)
            {
                return "";
            }
            else
            {
                object obj = Decode(val);
                if (obj != null)
                    return obj.ToString();
                else
                    return "";
            }
        }

    }

}